agent/remote_exec_test.go

```diff
@@ -107,6 +107,15 @@ func TestRemoteExecGetSpec_ACLToken(t *testing.T) {
 	testRemoteExecGetSpec(t, cfg)
 }
 
+func TestRemoteExecGetSpec_ACLAgentToken(t *testing.T) {
+	t.Parallel()
+	cfg := TestConfig()
+	cfg.ACLDatacenter = "dc1"
+	cfg.ACLAgentToken = "root"
+	cfg.ACLDefaultPolicy = "deny"
+	testRemoteExecGetSpec(t, cfg)
```

Comment:
I am not sure if this this testing anything here. It is passing this config struct to `testRemoteExecGetSpec`, but that doesn't seem to get passed to the test agent it creates there. What exactly is the intention of this `TestRemoteExecGetSpec_ACLAgentToken` test case?